trieste/models/gpflux/architectures.py

```diff
@@ -0,0 +1,301 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""
+This file contains wrappers for some implementations of basic GPflux architectures.
+"""
+
+from __future__ import annotations
+
+from typing import Optional, Tuple
+
+import gpflow.mean_functions
+import numpy as np
+import tensorflow as tf
+import tensorflow_probability as tfp
+from gpflow import Parameter
+from gpflow.utilities.bijectors import positive
+from gpflux.architectures import Config, build_constant_input_dim_deep_gp
+from gpflux.encoders import DirectlyParameterizedNormalDiag
+from gpflux.helpers import (
+    construct_basic_inducing_variables,
+    construct_basic_kernel,
+    construct_mean_function,
+)
+from gpflux.layers import GPLayer, GIGPLayer, LatentVariableLayer, LikelihoodLayer
+from gpflux.models import DeepGP, GIDeepGP
+from gpflux.types import ObservationType
+from scipy.cluster.vq import kmeans2
+
+from ...types import TensorType
+
+
+def build_vanilla_deep_gp(
+    X: TensorType,
```

Comment:
for consistency with the rest of the toolbox we should use `query_points` for inputs, instead of `X` here